snoop/data/filesystem.py

```diff
@@ -139,3 +146,47 @@ def create_file(parent_directory, name, original):
         ),
     )
     create_directory_children(fake_root, archive_listing_data)
+
+
+@shaorma('filesystem.create_attachment_files')
+def create_attachment_files(file_pk, email_parse):
+    def iter_parts(email_data):
+        yield email_data
+        for part in email_data.get('parts') or []:
+            yield from iter_parts(part)
+
+    with email_parse.open() as f:
+        email_data = json.load(f)
+
+    attachments = []
+    for part in iter_parts(email_data):
+        part_attachment = part.get('attachment')
+        if part_attachment:
+            attachments.append(part_attachment)
+
+    if attachments:
+        now = timezone.now()
+        file = models.File.objects.get(pk=file_pk)
+        (attachments_dir, _) = file.child_directory_set.get_or_create(
+            name='',
+            defaults=dict(
+                collection=file.collection,
+            ),
+        )
+        for attachment in attachments:
+            original = models.Blob.objects.get(pk=attachment['blob_pk'])
+            size = original.path().stat().st_size
+            now = timezone.now()
+
+            (file, _) = attachments_dir.child_file_set.get_or_create(
+                name=attachment['name'],
+                defaults=dict(
+                    collection=file.collection,
+                    ctime=now,
```

Comment:
I think these (`ctime`, `mtime`) should be a time taken from the email, not the current digest time. Or, if we can figure out their actual create/modify times from the email itself? If we can't figure them out, then just inherit them from the email file itself (the first one? the last one?)